node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/NodesResponse.java

```diff
@@ -246,4 +233,29 @@ private String lastElement(String path) {
         return path.substring(lastSlash+1);
     }
 
+    static Optional<String> nodeArchiveUrl(FlagSource flagSource, Node node) {
+        String bucket = Flags.SYNC_HOST_LOGS_TO_S3_BUCKET.bindTo(flagSource)
+                .with(FetchVector.Dimension.NODE_TYPE, node.type().name())
+                .with(FetchVector.Dimension.APPLICATION_ID, node.allocation().map(alloc -> alloc.owner().serializedForm()).orElse(null))
+                .value();
+        if (bucket.isBlank()) return Optional.empty();
+
+        StringBuilder sb = new StringBuilder(100).append("s3://").append(bucket).append('/');
+        if (node.type() == NodeType.tenant) {
+            if (node.allocation().isEmpty()) return Optional.empty();
+            ApplicationId app = node.allocation().get().owner();
+
+            sb.append(app.tenant().value()).append('/').append(app.application().value()).append('/').append(app.instance().value()).append('/');
+        } else {
+            sb.append("hosted-vespa/");
```

Comment:
Just using `hosted-vespa` for all non-tenants since
1. It is a reserved name, so there will be no real tenant with that name
2. Infrastructure nodes can be without allocation for some time and we still might want to have logs/core dumps from it